### DISCUSSION TITLE
Migrating Network Emulation from CORE v4.8 to Newer Version

### DISCUSSION THREAD

**Discussion by faceb912** [coreemu/core#796] [Q&A]
Hello,

I've been using CORE Network Emulator version 4.8 for a while now, and have developed a comprehensive network emulation that works perfectly on this version. I'm interested in migrating my project to the newer version of CORE.

However, I've encountered some compatibility issues while trying to do so. I attempted to export my current configuration via the XML file, but it appears there are some incompatibilities when I try to load it into the new version. The new version doesn't seem to accept or correctly interpret the XML file from the old version.

Would you be able to provide any guidance or suggestions on how to proceed with this migration? Are there any specific steps that I should follow, or potential pitfalls to avoid? Additionally, is there any detailed documentation on migrating projects between versions?

Any help on this matter would be greatly appreciated.

**Comment by bharnden** [comment]
Without knowing the issues you are having specifically, it is hard to say anything. But, the transition from the old GUI to new python based GUI did come with the expected loss of the GUI customization aspects. The general nodes, links, and their positions ideally should be translating. Then you would just need to setup the visuals again.

**Comment by cspiker7** [comment]
Each version lists what changed, but there is no detailed documentation on exactly what you need to update in the code to be compatible with the latest version. We've done several updates from 4.8 to newer versions with our older scenarios. Usually, the .imn to .xml works, but there might be manual changes you have to make to custom services. Feel free to list some errors and we can help you work through them.
